java/core/src/test/java/com/microsoft/bond/testmodel/MyEnum.java

```diff
@@ -78,6 +78,30 @@ public static MyEnum get(int value) {
         }
     }
 
+    public static MyEnum parse(String str) {
+        if (str == null) {
+            throw new IllegalArgumentException("Argument 'str' must not be null.");
+        } else if (str.equals("White")) {
+            return White;
+        } else if (str.equals("Yellow")) {
+            return Yellow;
+        } else if (str.equals("Orange")) {
+            return Orange;
+        } else if (str.equals("Purple")) {
+            return Purple;
+        } else if (str.equals("Blue")) {
+            return Blue;
+        } else if (str.equals("Green")) {
+            return Green;
+        } else if (str.equals("Brown")) {
+            return Brown;
+        } else if (str.equals("Black")) {
+            return Black;
+        } else {
+            throw new IllegalArgumentException("Invalid 'MyEnum' enum value: '" + str + "'.");
```

Comment:
Small point, but I don't think that applies here - the behavior you're talking about is a promise about pulling enums off the wire, if you let bond decide how to {,de}serialize them. If you've gone out of your way to serialize them as strings and then tried to get enums back through this function, you have done something weird and out of the bounds of the promise.